hamilton/function_modifiers/macros.py

```diff
@@ -773,65 +809,152 @@ def final_result(upstream_int: int) -> int:
                 return upstream_int
 
         In all likelihood, you should not be using this, and this is only here in case you want to expose a node for
-        consumption/output later. Setting the namespace in individual nodes as well as in ``pipe`` is not yet supported.
+        consumption/output later. Setting the namespace in individual nodes as well as in ``pipe_input`` is not yet supported.
+
     """
 
     def __init__(
         self,
         *transforms: Applicable,
         namespace: NamespaceType = ...,
+        on_input: base.TargetType = None,
         collapse=False,
         _chain=False,
     ):
         """Instantiates a ``@pipe_input`` decorator.
 
         :param transforms: step transformations to be applied, in order
         :param namespace: namespace to apply to all nodes in the pipe. This can be "..." (the default), which resolves to the name of the decorated function, None (which means no namespace), or a string (which means that all nodes will be namespaced with that string). Note that you can either use this *or* namespaces inside ``pipe_input()``...
+        :param on_input: setting the target parameter for all steps in the pipe. Leave empty to select only the first argument.
         :param collapse: Whether to collapse this into a single node. This is not currently supported.
         :param _chain: Whether to chain the first parameter. This is the only mode that is supported. Furthermore, this is not externally exposed. ``@flow`` will make use of this.
         """
+        base.NodeTransformer._early_validate_target(target=on_input, allow_multiple=True)
+
         self.transforms = transforms
         self.collapse = collapse
+        self.target = on_input
         self.chain = _chain
         self.namespace = namespace
 
+        if isinstance(on_input, str):  # have to do extra since strings are collections in python
+            self.target = [on_input]
+        elif isinstance(on_input, Collection):
```

Comment:
What's the use-case for a collection here? It's interesting, but feels a bit more complex than necessary.